### ISSUE TITLE
[Question] How can I update _settings?

### ISSUE THREAD

**Issue by VojtechVitek** [olivere/elastic#206]
Hi,
first of all, thanks for really nice and robust client.

I tried to find a way how to update index _settings via v2 or v3 API - unsuccessfully. Is there any way how could I update _settings via this client?

For example, I'd like to set `"dynamic": false` or `"dynamic": "string"` settings of my indices:

`curl -XPUT --data '{"index.mapper.dynamic":false}' localhost:9200/_settings`

Best,
- Vojtech

**Comment by olivere** [comment]
The [Update Indices Settings API](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-update-settings.html) is currently not implemented. I will add it over the next days. It's very straightforward to do so.

If you cannot wait, you can always execute any service manually by calling the [`PerformRequest`](https://github.com/olivere/elastic/blob/release-branch.v3/client.go#L944) func on a `Client`. Here's an example:

```go
client, err := elastic.NewClient(...)
if err != nil { ... }

// Invoke PUT /_settings with specific body
params := url.Values{}
body := `{"index.mapper.dynamic":false}`
_, err = client.PerformRequest("PUT", "/_settings", params, body)
if err != nil { ... }
```

**Comment by olivere** [comment]
Hey. I just merged the API and will release 3.0.19 soon. I will back-port it to elastic.v2 as well.

[See here](https://github.com/olivere/elastic/blob/release-branch.v3/indices_put_settings_test.go#L41) for an example of how to use it. Unfortunately, with the Get Settings API you have to deal with a generic `map[string]interface{}` as Elasticsearch does not have a typed interface for the settings and also simply returns a dictionary.

**Comment by VojtechVitek** [comment]
Thanks, this is awesome, @olivere.
